lib/api/model/model.dart

```diff
@@ -328,6 +328,7 @@ class ZulipStream {
 
   final int streamId;
   String name;
+  bool isArchived;
```

Comment:
We should make this optional. On older servers, this field will not be sent.

> is_archived: boolean
> A boolean indicating whether the channel is [archived](https://zulip.com/help/archive-a-channel).
> Changes: New in Zulip 10.0 (feature level 315). Previously, this endpoint never returned archived channels.

A clean way to do this might be defaulting this to `false`, since its value can be indicated from the fact that no archived channels were returned from this endpoint before, with a `// TODO(server-10) remove default` comment.